framework/src/service/ServiceHooks.cpp

```diff
@@ -75,10 +74,8 @@ void ServiceHooks::RemovedService(const ServiceReference<ServiceListenerHook>& /
 
 void ServiceHooks::Open()
 {
-  {
-    auto l = this->Lock(); US_UNUSED(l);
-    listenerHookTracker.reset(new ServiceTracker<ServiceListenerHook>(GetBundleContext(), this));
-  }
+  auto l = this->Lock(); US_UNUSED(l);
+  listenerHookTracker.reset(new ServiceTracker<ServiceListenerHook>(GetBundleContext(), this));
```

Comment:
The only reason I could find to ```reset()``` the ```listenerHookTracker``` each time was as an alternative to finding a place to instantiate ```listenerHookTracker``` once. Instantiating ```listenerHookTracker``` in the ```ServiceHooks``` constructor would have been ill-advised since ```listenerHookTracker``` is instantiated with the ```ServiceHooks``` ```this``` pointer.